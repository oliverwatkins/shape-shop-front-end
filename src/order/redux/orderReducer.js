import {Actions} from './productActions';
import type {OrderState} from "../../AppState";
import {DeliveryType, PaymentType} from "../../constants";

export function reducer(state: OrderState = initialState , action) {
	switch (action.type) {
		case Actions.UPDATE_ADDRESS:
			return {
				...state,
				addressEntity: action.value
			}
		case Actions.UPDATE_CREDIT_CARD:

			let cardType = action.value.paymentMethod.card.brand;


			let m = action.value.paymentMethod.card.exp_month;
			let y = action.value.paymentMethod.card.exp_year;

			let last4 = action.value.paymentMethod.card.last4;

			return {
				...state,
				creditCardEntity: {
					type: cardType,
					expDate: "" + m + y,
					number: "xxxx-xxxxx-xxxxx-" + last4
				}
			}
		case Actions.UPDATE_PAYMENT_TYPE:
			return {
				...state,
				paymentType: action.value
			}
		case Actions.UPDATE_DELIVERY_TYPE:
			return {
				...state,
				deliveryType: action.value
			}
		case Actions.PLACE_ORDER:
			return {
				...state,
				submittingOrder: true,
			}
		case Actions.PLACE_ORDER_SUCCESS:
			return {
				...state,
				submittingOrder: false,
			}
		case Actions.PLACE_ORDER_ERROR:
			return {
				...state,
				submittingOrder: false,
				orderError: action.value,
			}
		default :
			return state;
	}
}

const initialState = {
	items: [],
	paymentType: PaymentType.cash,
	deliveryType: DeliveryType.pickup,
};
