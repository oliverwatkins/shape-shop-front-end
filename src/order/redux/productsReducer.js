import {Actions} from './productActions';

import type {ProductsState} from "../../AppState";
import {Notify} from "../../notify";


export function reducer(state:ProductsState = initialState, action) {
	switch (action.type) {

		case Actions.FETCH_PRODUCTS_SUCCESS:
			return {
				...state,
				items: action.data,
			};

		case Actions.FETCH_PRODUCTS_ERROR:

			Notify.error(action.errorMsg)

			return {
				...state,
				items: [],
				productsError: "Error Getting Products"
			};

		case Actions.UPDATE_PRODUCT_SELECTION:

			let foundProd = state.items.filter(item => item.id === action.id);

			let fp = foundProd[0];

			fp.quantity = action.value;

			return {
				...state,
				items: state.items.map((item) => {
					if (item.id === action.id) {
						//TODO should spread
						return {"id": item.id,
							"name": item.name,
							"price": item.price,
							"type": item.type,
							"quantity": action.value,
							"imageFilename": item.imageFilename,
						}
					}
					return item;
				}),
			};

		default :
			return state;
	}
}

const initialState = {
	items: []

	// products: {
	// 	items: []
	// }
};
