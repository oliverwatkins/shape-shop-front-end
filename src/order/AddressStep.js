import * as React from 'react';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faTruck} from "@fortawesome/free-solid-svg-icons";
import {Formik} from "formik";

import "./order.scss"
import {Redirect} from "react-router";
import {createUpdateAddress, createUpdateDeliveryType} from "./redux/productActions";
import {connect} from "react-redux";
import {wizardPages as pages} from "./OrderWizard"
import {NextButton} from "./buttons/NextButton";
import {BackButton} from "./buttons/BackButton";
import {DeliveryType} from "../constants";

type Props = {
	updateAddress: (value, id)=>void,
	updateDeliveryType: (value)=>void
}

type State = {
	redirect: boolean,
	deliveryType: "DELIVERY" | "PICKUP",
}

export class Address extends React.PureComponent<Props, State> {

	constructor() {
		super();
		this.state = {
			redirect: false,
			deliveryType: DeliveryType.pickup,
		}
	}


	setRedirect() {
		this.setState({
			redirect: true
		})
	}

	onRadioChanged = (e) => {
		// throw "errored"
		this.setState({
			deliveryType: e.currentTarget.value,
		});
		this.props.updateDeliveryType(e.currentTarget.value)
	}

	render() {
		if (this.state.redirect) {
			return <Redirect to={pages.WHICH_PAYMENT}/>
		}


		return (
			<div className="wizardPanel address-step">

				<h2 className={"wizardHeader"}>Delivery or Pickup?</h2>
				<div className="wizardMain">
					<BackButton page={pages.DRINK_LIST}/>
					<div className="wizardCenter">

						<div className="icon-container">
							<FontAwesomeIcon icon={faTruck} style={{fontSize: "60px", color: "navy", margin: "25px"}}/>
						</div>

						<div>
							<input type="radio"
										 id="contactChoice1"
										 name="pckupOrDelivery"
										 value={DeliveryType.pickup}
										 onChange={this.onRadioChanged}
										 checked={this.state.deliveryType === DeliveryType.pickup}/>
							<label htmlFor="contactChoice1">Pickup</label>
							<input type="radio"
										 id="contactChoice2"
										 name="pckupOrDelivery"
										 value={DeliveryType.delivery}
										 checked={this.state.deliveryType === DeliveryType.delivery}
										 onChange={this.onRadioChanged}/>
							<label htmlFor="contactChoice2">Delivery</label>
						</div>

						{this.state.deliveryType && (this.state.deliveryType === DeliveryType.delivery) &&
						<Formik
							initialValues={{email: '', password: '', name: ''}}
							validate={validator}
							onSubmit={(values, blah) => {
								setTimeout(() => {
									// alert(JSON.stringify(values, null, 2));
									blah.setSubmitting(false);

									this.props.updateAddress(values);
									this.setRedirect()
								}, 400);
							}}>

							{({values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting}) => (
								<form onSubmit={handleSubmit} id="addressForm">
									<div>
										<label htmlFor="name">Name</label>
										<input
											id="name"
											type="text"
											name="name"
											onChange={handleChange}
											onBlur={handleBlur}
											value={values.name}
										/>
										<span className={"error"}>
									{errors.name && touched.name && errors.name}
									</span>
									</div>
									<div>
										<label htmlFor="street">Strasse</label>
										<input
											id="street"
											type="text"
											name="street"
											onChange={handleChange}
											onBlur={handleBlur}
											value={values.street}
										/>
										<span className={"error"}>
									{errors.street && touched.street && errors.street}
									</span>
									</div>
									<div>
										<label htmlFor="postcode">Postleitzahl</label>
										<input
											id="postcode"
											type="text"
											name="postcode"
											onChange={handleChange}
											onBlur={handleBlur}
											value={values.postcode}
										/>
										<span className={"error"}>
									{errors.postcode && touched.postcode && errors.postcode}
									</span>
									</div>
									<div>
										<label htmlFor="tel">Telefon</label>
										<input
											id="tel"
											type="text"
											name="telephone"
											onChange={handleChange}
											onBlur={handleBlur}
											value={values.telephone}
										/>
										<span className={"error"}>
									{errors.telephone && touched.telephone && errors.telephone}
									</span>
									</div>
									<div>
										<label htmlFor="email">Email</label>
										<input
											id="email"
											type="email"
											name="email"
											onChange={handleChange}
											onBlur={handleBlur}
											value={values.email}
										/>
										<span className={"error"}>
									{errors.email && touched.email && errors.email}
									</span>
									</div>
									<button type="submit" disabled={isSubmitting}>
										Submit
									</button>
								</form>
							)}
						</Formik>
						}
					</div>
					{this.state.deliveryType === DeliveryType.delivery &&
					<NextButton label={"NEXT"} type={"submit"} form={"addressForm"}/>}

					{(this.state.deliveryType === DeliveryType.pickup) &&
					<NextButton label={"NEXT"} page={pages.WHICH_PAYMENT}/>}
				</div>
			</div>
		);
	}
}


let validator = (values) => {
	const errors = {};

	if (!values.name) {
		errors.name = 'Name Required';
	}

	if (!values.email) {
		errors.email = 'Required';
	} else if (
		!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
	) {
		errors.email = 'Invalid email address';
	}
	return errors;
}


const mapDispatchToProps = dispatch => {
	return {
		updateAddress: (value, id) => {
			dispatch(createUpdateAddress(value, id));
		},
		updateDeliveryType: (value) => {
			dispatch(createUpdateDeliveryType(value));
		},
	};
};

export default connect(
	null,
	mapDispatchToProps,
)(Address);