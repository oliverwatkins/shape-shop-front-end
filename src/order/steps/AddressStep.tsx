import * as React from 'react';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faTruck} from "@fortawesome/free-solid-svg-icons";

import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from "yup";

import {Redirect} from "react-router";
import {useDispatch, useSelector} from "react-redux";
import {wizardPages as pages} from "../OrderWizardContainer"
import {NextButton} from "../buttons/NextButton";
import {BackButton} from "../buttons/BackButton";
import {Address, DeliveryType, Product} from "../../AppState";
import {updateAddressAction, updateDeliveryTypeAction} from "../../admin/redux/orderReducer";
import {useForm} from "react-hook-form";
import {selectOrder} from "../../selectors";

import "./addressStep.scss"
import Header from "./Header";

export default function AddresStep() {

    const order = useSelector(selectOrder);

    let schema = yup.object({
        name: yup.string().required(),
        telephone: yup.number().positive().integer().required(),
        email: yup.string().email(),
        postcode: yup.string().required(),
        street: yup.string().required(),
    }).required();

    let schema2 = yup.object({
            name: yup.string().required(),
            telephone: yup.number().positive().integer().required(),
            email: yup.string().email(),
        }).required();

    const { register, handleSubmit, formState } = useForm<Address>({
        resolver: yupResolver((order.deliveryType === DeliveryType.DELIVERY) ? schema: schema2),
        defaultValues: {
            name: order.address ? order.address.name : "",
            email: order.address ? order.address.email : "",
            street: order.address ? order.address.street : "",
            telephone: order.address ? order.address.telephone : "",
            postcode: order.address ? order.address.postcode : ""
        }
    });

    let errors = formState.errors

    const dispatch = useDispatch();
    const [redirect, setRedirect] = React.useState<boolean>(false);

    function onRadioChanged(e: any) {
        dispatch(updateDeliveryTypeAction(e.currentTarget.value));
    }

    if (redirect) {
        return <Redirect to={pages.WHICH_PAYMENT}/>
    }

    const onSubmit = (data: Address) => {
        dispatch(updateAddressAction({address: data}));
        setRedirect(true)
    }
    return (
        <div className="wizardPanel address-step">

            {/*{order && order.deliveryType}*/}
            <Header text={"asdf"}/>
            {/*<h2 className={"wizardHeader"}>Delivery or Pickup?</h2>*/}
            <div className="wizardMain">
                <BackButton page={"/order/cat_drinks"}/>
                <form onSubmit={handleSubmit(onSubmit)} id="addressForm" aria-label="form">
                    <div className="wizardCenter">
                        <div className="icon-container">
                            <FontAwesomeIcon icon={faTruck} style={{fontSize: "60px", color: "navy", margin: "25px"}}/>
                        </div>
                        <div className={"radioBox"}>
                            <input type="radio"
                                   id="contactChoice1"
                                   name="pickupOrDelivery"
                                   value={DeliveryType.PICKUP}
                                   onChange={onRadioChanged}
                                   checked={order.deliveryType === DeliveryType.PICKUP}/>
                            <label htmlFor="contactChoice1">Pickup</label>
                            <input type="radio"
                                   id="contactChoice2"
                                   name="pickupOrDelivery"
                                   value={DeliveryType.DELIVERY}
                                   checked={order.deliveryType === DeliveryType.DELIVERY}
                                   onChange={onRadioChanged}/>
                            <label htmlFor="contactChoice2">Delivery</label>
                        </div>

                        <div>
                            <label htmlFor="name">Name</label>
                            <input id="name" type="text"
                                {...register("name", {required: true, maxLength: 45, })}
                            />
                            <span className={"error"}>
                                {errors.name?.type === 'required' && <p role="alert">Name is required</p>}
                            </span>
                        </div>
                        {order.deliveryType && (order.deliveryType === DeliveryType.DELIVERY) &&
                            <div>
                                <div>
                                    <label htmlFor="street">Strasse</label>
                                    <input id="street" type="text"
                                        {...register("street", {required: true, maxLength: 85})}
                                    />
                                    <span className={"error"}></span>
                                </div>
                                <div>
                                    <label htmlFor="postcode">Postleitzahl</label>
                                    <input
                                        id="postcode"
                                        type="text"
                                        {...register("postcode", {required: true, maxLength: 85})}
                                    />
                                    <span className={"error"}></span>
                                </div>
                            </div>
                        }
                        <div>
                            <label htmlFor="tel">Telefon</label>
                            <input
                                id="tel"
                                type="text"
                                {...register("telephone", {required: true, maxLength: 85})}
                            />
                            <span className={"error"}>
                                {errors.telephone?.type === 'required' && <p role="alert">Telephone is required</p>}
                            </span>
                        </div>
                        <div>
                            <label htmlFor="email">Email</label>
                            <input
                                id="email"
                                type="text"
                                value={order.address?.email}
                                {...register("email", {required: false, maxLength: 85})}
                            />
                            <span className={"error"}>
                                {errors.email?.type === 'pattern' && <p role="alert">Email pattern is wrong</p>}
                            </span>
                        </div>
                    </div>
                </form>
                <NextButton label={"next"} type={"submit"} form={"addressForm"} disabled={false}/>
            </div>
        </div>
    );
}

let validator = (values: any) => {
    const errors: any = {};

    if (!values.name) {
        errors.name = 'Name Required';
    }

    if (!values.email) {
        errors.email = 'Required';
    } else if (
        !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
    ) {
        errors.email = 'Invalid email address';
    }
    return errors;
}
