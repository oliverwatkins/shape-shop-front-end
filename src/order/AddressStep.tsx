import * as React from 'react';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faTruck} from "@fortawesome/free-solid-svg-icons";

import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from "yup";

import "./order.scss"
import {Redirect} from "react-router";
import {useDispatch} from "react-redux";
import {wizardPages as pages} from "./OrderWizardContainer"
import {NextButton} from "./buttons/NextButton";
import {BackButton} from "./buttons/BackButton";
import {Address, DeliveryType, Product} from "../AppState";
import {updateAddressAction, updateDeliveryTypeAction} from "../admin/redux/orderReducer";
import {useForm} from "react-hook-form";
import {Link} from "react-router-dom";

const schema = yup.object({
    name: yup.string().required(),
    telephone: yup.number().positive().integer().required(),
    email: yup.string().email()
}).required();


export default function AddresStep() {

    const { register, handleSubmit, formState } = useForm<Address>({
        resolver: yupResolver(schema)
    });

    console.info("errors")

    let errors = formState.errors

    try {
        console.info(JSON.stringify(errors))
    }
    catch(e) {
        console.error("circular error")
    }

    const dispatch = useDispatch();
    const [redirect, setRedirect] = React.useState<boolean>(false);
    const [deliveryType, setDeliveryType] = React.useState(DeliveryType.pickup);

    function onRadioChanged(e: any) {
        setDeliveryType(e.currentTarget.value)
        dispatch(updateDeliveryTypeAction(e.currentTarget.value));
    }

    if (redirect) {
        return <Redirect to={pages.WHICH_PAYMENT}/>
    }

    const onSubmit = (data: Address) => {
        // alert(JSON.stringify(data))

        dispatch(updateAddressAction({address: data}));
        setRedirect(true)
    }
    return (
        <div className="wizardPanel address-step">

            {formState.isValid && <h4>IS VALID</h4> }
            <h2 className={"wizardHeader"}>Delivery or Pickup?</h2>
                <div className="wizardMain">
                    <BackButton page={"/order/cat_drinks"}/>
                    <form onSubmit={handleSubmit(onSubmit)} id="addressForm" aria-label="form" style={{border: "5px green dashed", width: "100%"}}>

                        <div className="wizardCenter">

                            <div className="icon-container">
                                <FontAwesomeIcon icon={faTruck}
                                                 style={{fontSize: "60px", color: "navy", margin: "25px"}}/>
                            </div>

                            <div className={"radioBox"}>
                                <input type="radio"
                                       id="contactChoice1"
                                       name="pickupOrDelivery"
                                       value={DeliveryType.pickup}
                                       onChange={onRadioChanged}
                                       checked={deliveryType === DeliveryType.pickup}/>
                                <label htmlFor="contactChoice1">Pickup</label>
                                <input type="radio"
                                       id="contactChoice2"
                                       name="pckupOrDelivery"
                                       value={DeliveryType.delivery}
                                       checked={deliveryType === DeliveryType.delivery}
                                       onChange={onRadioChanged}/>
                                <label htmlFor="contactChoice2">Delivery</label>
                            </div>

                            <div>
                                <label htmlFor="name">Name</label>
                                <input
                                    id="name"
                                    type="text"
                                    {...register("name", {required: true, maxLength: 45})}
                                />
                                <span className={"error"}>
                                    {errors.name?.type === 'required' && <p role="alert">Name is required</p>}
                            </span>
                            </div>
                            {deliveryType && (deliveryType === DeliveryType.delivery) &&
                                <div>
                                    <div>
                                        <label htmlFor="street">Strasse</label>
                                        <input
                                            id="street"
                                            type="text"
                                            {...register("street", {required: false, maxLength: 85})}
                                        />
                                        <span className={"error"}>

                            </span>
                                    </div>
                                    <div>
                                        <label htmlFor="postcode">Postleitzahl</label>
                                        <input
                                            id="postcode"
                                            type="text"
                                            {...register("postcode", {required: false, maxLength: 85})}
                                        />
                                        <span className={"error"}>
                                </span>
                                    </div>
                                </div>
                            }
                            <div>
                                <label htmlFor="tel">Telefon</label>
                                <input
                                    id="tel"
                                    type="text"
                                    {...register("telephone", {required: true, maxLength: 85})}
                                />
                                <span className={"error"}>
                                    {errors.telephone?.type === 'required' && <p role="alert">Telephone is required</p>}
                                </span>
                            </div>
                            <div>
                                <label htmlFor="email">Email</label>
                                <input
                                    id="email"
                                    type="text"
                                    {...register("email", {required: false, maxLength: 85})}
                                />
                                <span className={"error"}>
                                    {errors.email?.type === 'required' && <p role="alert">Email is required</p>}
                                </span>
                            </div>
                        </div>
                    </form>
                <NextButton label={"next"} type={"submit"} form={"addressForm"} disabled={false}/>
            </div>
        </div>
    );
}

let validator = (values: any) => {
    const errors: any = {};

    if (!values.name) {
        errors.name = 'Name Required';
    }

    if (!values.email) {
        errors.email = 'Required';
    } else if (
        !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
    ) {
        errors.email = 'Invalid email address';
    }
    return errors;
}
