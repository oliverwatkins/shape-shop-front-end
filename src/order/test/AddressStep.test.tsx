import React from 'react';
import {Provider} from 'react-redux';
import AddressStep from "../steps/AddressStep";
import {MemoryRouter} from "react-router-dom";

import {fireEvent, render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'
import {combineReducers, createStore} from "redux";
import {productsReducer as products} from "../../admin/redux/productsReducer";
import {reducer as order} from "../../admin/redux/orderReducer";
import {reducer as admin} from "../../admin/redux/adminReducer";
import {getMockData} from "./mockData";
import {reducer as loginReducer} from "../../login/redux/loginReducer";

// export function createTestStore() {
//
// 	return createStore(
// 		combineReducers({
// 			// @ts-ignore POS ts comnpiler
// 			login: loginReducer,
// 			products: products,
// 			order: order,
// 			admin: admin
// 		}), getMockData()
// 	);
// }
export function createTestStore() {
	return createStore(
		combineReducers({
			// @ts-ignore POS ts comnpiler
			login: loginReducer,
			products: products,
			order: order,
			admin: admin
		}), getMockData()
	);
}
//TODO change all other enzyme tests to testing library (like here)
describe('Address Step test', () => {
	const store = createTestStore();
	it('matches snapshot', () => {
		expect(render(<Provider store={store}>
			<MemoryRouter>
				<AddressStep/>
			</MemoryRouter>
		</Provider>)).toMatchSnapshot();
	});


	it('select radio button - check that it works', async () => {

		render(<Provider store={store}>
		 			<MemoryRouter>
		 				<AddressStep/>
					</MemoryRouter>
				</Provider>)

		await screen.findAllByRole('heading')

		let x = screen.getAllByRole('heading')

		expect(x[0]).toHaveTextContent('Delivery or Pickup?')
		expect(x[1]).toHaveTextContent('Order Summary')

		const labelRadio: HTMLInputElement[] = await screen.findAllByRole('radio');

		expect(labelRadio[0].checked).toEqual(true);

		let firstRadio = labelRadio[0];
		let secondRadio = labelRadio[1];

		fireEvent.click(firstRadio);

		let textBoxes: HTMLInputElement[] = await screen.findAllByRole('textbox');
		expect(textBoxes.length).toBe(3);

		fireEvent.click(secondRadio);

		textBoxes = await screen.findAllByRole('textbox');
		expect(textBoxes.length).toBe(5);

		fireEvent.click(firstRadio);

		textBoxes = await screen.findAllByRole('textbox');
		expect(textBoxes.length).toBe(3);
	});
});
