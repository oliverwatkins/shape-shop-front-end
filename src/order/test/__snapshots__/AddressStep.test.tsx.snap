// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Step test matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wiz-container address-step"
      >
        <div
          class="emptySpace"
        />
        <div
          class="wizardHeader wiz-item"
        >
          <h3>
            Delivery or Pickup?
          </h3>
        </div>
        <a
          href="/order/cat_drinks"
        >
          <button
            class="backButton next-prev-button wiz-item"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-circle-left fa-w-16 icon"
              data-icon="arrow-circle-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
                fill="currentColor"
              />
            </svg>
          </button>
        </a>
         
        <div
          class="wizardStep"
        >
          <form
            aria-label="form"
            id="addressForm"
          >
            <div
              class="icon-container"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-truck fa-w-20 "
                data-icon="truck"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="font-size: 60px; color: navy; margin: 25px;"
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M624 352h-16V243.9c0-12.7-5.1-24.9-14.1-33.9L494 110.1c-9-9-21.2-14.1-33.9-14.1H416V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h16c0 53 43 96 96 96s96-43 96-96h128c0 53 43 96 96 96s96-43 96-96h48c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM160 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm320 0c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-208H416V144h44.1l99.9 99.9V256z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="radioBox"
            >
              <input
                checked=""
                id="contactChoice1"
                name="pickupOrDelivery"
                type="radio"
                value="PICKUP"
              />
              <label
                for="contactChoice1"
              >
                Pickup
              </label>
              <input
                id="contactChoice2"
                name="pickupOrDelivery"
                type="radio"
                value="DELIVERY"
              />
              <label
                for="contactChoice2"
              >
                Delivery
              </label>
            </div>
            <div>
              <label
                for="name"
              >
                Name
              </label>
              <input
                id="name"
                name="name"
                type="text"
              />
              <span
                class="error"
              />
            </div>
            <div>
              <label
                for="tel"
              >
                Telefon
              </label>
              <input
                id="tel"
                name="telephone"
                type="text"
              />
              <span
                class="error"
              />
            </div>
            <div>
              <label
                for="email"
              >
                Email
              </label>
              <input
                id="email"
                name="email"
                type="text"
                value=""
              />
              <span
                class="error"
              />
            </div>
          </form>
        </div>
        <button
          class="nextButton next-prev-button wiz-item"
          form="addressForm"
          type="submit"
        >
          <div
            style="display: flex;"
          >
            <div
              class="label"
            >
              next
            </div>
            <div>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-circle-right fa-w-16 icon"
                data-icon="arrow-circle-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </button>
        <div
          class="order-summary wiz-item"
          data-testid="order-summary"
        >
          <h3>
            Order Summary
          </h3>
          <div
            style="padding-left: 7px;"
          >
            Nothing selected
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wiz-container address-step"
    >
      <div
        class="emptySpace"
      />
      <div
        class="wizardHeader wiz-item"
      >
        <h3>
          Delivery or Pickup?
        </h3>
      </div>
      <a
        href="/order/cat_drinks"
      >
        <button
          class="backButton next-prev-button wiz-item"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-circle-left fa-w-16 icon"
            data-icon="arrow-circle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 504C119 504 8 393 8 256S119 8 256 8s248 111 248 248-111 248-248 248zm28.9-143.6L209.4 288H392c13.3 0 24-10.7 24-24v-16c0-13.3-10.7-24-24-24H209.4l75.5-72.4c9.7-9.3 9.9-24.8.4-34.3l-11-10.9c-9.4-9.4-24.6-9.4-33.9 0L107.7 239c-9.4 9.4-9.4 24.6 0 33.9l132.7 132.7c9.4 9.4 24.6 9.4 33.9 0l11-10.9c9.5-9.5 9.3-25-.4-34.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </a>
       
      <div
        class="wizardStep"
      >
        <form
          aria-label="form"
          id="addressForm"
        >
          <div
            class="icon-container"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-truck fa-w-20 "
              data-icon="truck"
              data-prefix="fas"
              focusable="false"
              role="img"
              style="font-size: 60px; color: navy; margin: 25px;"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M624 352h-16V243.9c0-12.7-5.1-24.9-14.1-33.9L494 110.1c-9-9-21.2-14.1-33.9-14.1H416V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h16c0 53 43 96 96 96s96-43 96-96h128c0 53 43 96 96 96s96-43 96-96h48c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM160 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm320 0c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-208H416V144h44.1l99.9 99.9V256z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="radioBox"
          >
            <input
              checked=""
              id="contactChoice1"
              name="pickupOrDelivery"
              type="radio"
              value="PICKUP"
            />
            <label
              for="contactChoice1"
            >
              Pickup
            </label>
            <input
              id="contactChoice2"
              name="pickupOrDelivery"
              type="radio"
              value="DELIVERY"
            />
            <label
              for="contactChoice2"
            >
              Delivery
            </label>
          </div>
          <div>
            <label
              for="name"
            >
              Name
            </label>
            <input
              id="name"
              name="name"
              type="text"
            />
            <span
              class="error"
            />
          </div>
          <div>
            <label
              for="tel"
            >
              Telefon
            </label>
            <input
              id="tel"
              name="telephone"
              type="text"
            />
            <span
              class="error"
            />
          </div>
          <div>
            <label
              for="email"
            >
              Email
            </label>
            <input
              id="email"
              name="email"
              type="text"
              value=""
            />
            <span
              class="error"
            />
          </div>
        </form>
      </div>
      <button
        class="nextButton next-prev-button wiz-item"
        form="addressForm"
        type="submit"
      >
        <div
          style="display: flex;"
        >
          <div
            class="label"
          >
            next
          </div>
          <div>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-circle-right fa-w-16 icon"
              data-icon="arrow-circle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm-28.9 143.6l75.5 72.4H120c-13.3 0-24 10.7-24 24v16c0 13.3 10.7 24 24 24h182.6l-75.5 72.4c-9.7 9.3-9.9 24.8-.4 34.3l11 10.9c9.4 9.4 24.6 9.4 33.9 0L404.3 273c9.4-9.4 9.4-24.6 0-33.9L271.6 106.3c-9.4-9.4-24.6-9.4-33.9 0l-11 10.9c-9.5 9.6-9.3 25.1.4 34.4z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </button>
      <div
        class="order-summary wiz-item"
        data-testid="order-summary"
      >
        <h3>
          Order Summary
        </h3>
        <div
          style="padding-left: 7px;"
        >
          Nothing selected
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
